src/pages/2018-11-21-bull/index.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Processing jobs across microservices using Bull 🐂
+date: '2018-11-21T22:12:03.284Z'
+path: '/job-processor-async/'
+---
+
+> An in-depth explanation into why we invested in the open source package, <a href="https://github.com/OptimalBits/bull" target="_blank">Bull</a>, even though it didn't quite fit our needs.
+
+Into the microservice world we go, exploring where no developer has gone before. When switching to a service oriented stack versus the ol' monolith, new problems arise. The simple async job processor of the past doesn't fit in this new architecture. We found <a href="https://github.com/OptimalBits/bull" target="_blank">Bull</a>, a Node.js package, to cover our needs, but needed tweaks to work in this new world.
+
+If you have one service which performs a task, e.g. an image processor, and another service which needs to queue jobs to it, we cannot use traditional technology. I am of course mentioning things like DelayedJob or Celery which you may be familiar with. Keeping it 2018, this Node.js package is lightning fast, built to work with Redis and has an active community. It didn't quite fit our needs at first as it processed jobs in the same app as the queue-mechanism. This is fine for traditional apps, but for our setup we needed to manage jobs **across systems**.
+
+<table class="image">
+  <caption align="bottom">This</caption>
+  <tr><td><img src="./bull-post1.png" alt="single-service-with-jobs"/></td></tr>
+</table>
+
+versus
+
+<table class="image">
+  <caption align="bottom">This</caption>
+  <tr><td><img src="./bull-post2.png" alt="multiple-services-job-queue"/></td></tr>
+</table>
+
+We needed to make this work in an async fashion where the worker may not be in the same repo or service as the service running Bull itself. Bull had a simple way to define the processing logic:
+
+```javascript
+queue.process(async () => {
+  doWork()
+})
+```
+
+This way, whenever a job came into a waiting queue, Bull knew how to process it and throw it to the completed queue. You may be thinking, "to work in this new fashion, how about we just don't define this method?", we tried this out, and it _worked!_. We now need to manage the queue manually, but that's to be expected. Forward into the weeds we go.
```

Comment:
I'm confused why the above method is not compatible with the "new fashion". What prevents us from using this method in the "new fashion"?